spec/visual/samples/squirrel

```diff
@@ -0,0 +1,406 @@
+
+local susparam = coro.call("test","coroutine"); //starts the coroutine
+
+local i = 1;
+do
+{
+    ::print("suspend passed ["+susparam+"]\n")
+    susparam = coro.wakeup("ciao "+i);
+    ++i;
+}while(coro.getstatus()=="suspended")
+
+::print("return passed ["+susparam+"]\n")
+
+###
+
+
+local base_vec={
+    function _add(n)
+    {
+        return {
+            x=x+n.x,
+            y=y+n.y,
+            z=z+n.z,
+        }
+    }
+    function _sub(n)
+    {
+        return {
+            x=x-n.x,
+            y=y-n.y,
+            z=z-n.z,
+        }
+    }
+    function _div(n)
+    {
+        return {
+            x=x/n.x,
+            y=y/n.y,
+            z=z/n.z,
+        }
+    }
+    function _mul(n)
+    {
+        return {
+            x=x*n.x,
+            y=y*n.y,
+            z=z*n.z,
+        }
+    }
+    function _modulo(n)
+    {
+        return {
+            x=x%n,
+            y=y%n,
+            z=z%n,
+        }
+    }
+    function _typeof() {return "vector";}
+    function _get(key)
+    {
+        if(key==100)
+        {
+            return test_field;
+        }
+    },
+    function _set(key,val)
+    {
+        ::print("key = "+key+"\n");
+        ::print("val = "+val+"\n")
+        if(key==100)
+        {
+            return test_field=val;
+        }
+    }
+    test_field="nothing"
+}
+
+function vector(_x,_y,_z)
+{
+    return {x=_x,y=_y,z=_z }.setdelegate(base_vec);
+}
+////////////////////////////////////////////////////////////
+
+local v1=vector(1.5,2.5,3.5);
+local v2=vector(1.5,2.5,3.5);
+
+local r=v1+v2;
+
+
+foreach(i,val in r)
+{
+    print(i+" = "+val+"\n");
+}
+
+r=v1*v2;
+
+foreach(i,val in r)
+{
+    print(i+" = "+val+"\n");
+}
+
+r=v1/v2;
+
+foreach(i,val in r)
+{
+    print(i+" = "+val+"\n");
+}
+
+r=v1-v2;
+
+foreach(i,val in r)
+{
+    print(i+" = "+val+"\n");
+}
+
+r=v1%2;
+
+foreach(i,val in r)
+{
+    print(i+" = "+val+"\n");
+}
+
+print(v1[100]+"\n");
+v1[100]="set SUCCEEDED";
+print(v1[100]+"\n");
+
+if(typeof v1=="vector")
+    print("<SUCCEEDED>\n");
+else
+    print("<FAILED>\n");
+
+###
+
+/*
+*Random number function from The Great Computer Language shootout
+*converted to a generator func
+*/
+
+function gen_random(max) {
+    local last=42
+    local IM = 139968;
+    local IA = 3877;
+    local IC = 29573;
+    for(;;){  //loops forever
+        yield (max * (last = (last * IA + IC) % IM) / IM);
+    }
+}
+
+local randtor=gen_random(100);
+
+print("RAND NUMBERS \n")
+
+for(local i=0;i<10;i+=1)
+    print(">"+resume randtor+"\n");
+
+print("FIBONACCI \n")
+function fiboz(n)
+{
+    local prev=0;
+    local curr=1;
+    yield 1;
+
+    for(local i=0;i<n-1;i+=1)
+    {
+        local res=prev+curr;
+        prev=curr;
+        yield curr=res;
+    }
+    return prev+curr;
+}
+
+foreach(val in fiboz(10))
+{
+    ::print(">"+val+"\n");
+}
+
+###
+
+
+PEntity <- {
+    name="noname"
+    pos={x=0,y=0,z=0}
+    type="entity"
+    //methamethod
+    _typeof=function()
```

Comment:
These parentheses are not matching and are being represented with error tokens if you have a look at the code by running `rackup` from the root of the repo.